### PR TITLE
2nd proofreading of api/v2/checluster_types.go (grammar, interpunction, clarity etc.)

### PR DESCRIPTION
**Pull request by deerskindoll**
### What does this PR do?
2nd proofreading of api/v2/checluster_types.go (grammar, interpunction, clarity etc.)

### Screenshot/screencast of this PR

image

### What issues does this PR fix or reference?
 
https://issues.redhat.com/browse/RHDEVDOCS-4053

### How to test this PR?

please check edited content

```bash
cat EOF << > /tmp/patch.yaml
<PATCH_CONTENT>
EOF

chectl server:deploy \
     --installer operator \
     --platform <PLATFORM_TO_DEPLOY> \
     --che-operator-image <CUSTOM_OPERATOR_IMAGE> \
     --che-operator-cr-patch-yaml /tmp/patch.yaml
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: api/v2/checluster_types.go [hunk 1]
```diff
@@ -103,24 +103,24 @@ type CheClusterComponents struct {
...
+	// Defines annotations which will be set for an Ingress (a route for OpenShift platform).
 	// The defaults for kubernetes platforms are:
 	//     kubernetes.io/ingress.class:                       "nginx"
 	//     nginx.ingress.kubernetes.io/proxy-read-timeout:    "3600",
 	//     nginx.ingress.kubernetes.io/proxy-connect-timeout: "3600",
 	//     nginx.ingress.kubernetes.io/ssl-redirect:          "true"
 	// +optional
 	Annotations map[string]string `json:"annotations,omitempty"`
-	// For an OpenShift cluster, the Operator uses the domain to generate a hostname for a route.
+	// For an OpenShift cluster, the operator uses the domain to generate a hostname for the route.
```
Comment (rolfedh) [inline api/v2/checluster_types.go:120]: From the [SSG wordlist for the letter O](https://redhat-documentation.github.io/supplementary-style-guide/#_o): The term "Operator" in the context of Kubernetes is always capitalized to distinguish it from other types of operators, such as human or mathematical operators.

File: api/v2/checluster_types.go [hunk 2]
```diff
@@ -103,24 +103,24 @@ type CheClusterComponents struct {
...
 	// The generated hostname follows this pattern: che-<che-namespace>.<domain>. The <che-namespace> is the namespace where the CheCluster CRD is created.
-	// In a conjunction with labels it creates a route, which is served by a non-default Ingress controller.
-	// For Kubernetes cluster it contains a global ingress domain. This MUST be explicitly specified: there are no defaults.
+	// In conjunction with labels, it creates a route served by a non-default Ingress controller.
+	// For a Kubernetes cluster, it contains a global Ingress domain. This MUST be specified, there are no default values.
```
Comment (rolfedh) [inline api/v2/checluster_types.go:123]: I searched the repo for "Ingress domain". This is the only instance that capitalizes "Ingress."

Comment (rolfedh) [reply api/v2/checluster_types.go:123]: - The antecedents for "it" and "This" are ambiguous. Consider mentioning what they refer to. For example, "This domain MUST be specified..." 
- Consider using the active voice: `<something> does not have a default value: you must specify its value.`

File: api/v2/checluster_types.go (hunk 1 above)
Comment (rolfedh) [reply api/v2/checluster_types.go:120]: This PR contains multiple instances of "operator."

File: api/v2/checluster_types.go [hunk 3]
```diff
@@ -228,10 +228,10 @@ type DevfileRegistry struct {
 // Configuration settings related to the database used by the Che installation.
 // +k8s:openapi-gen=true
 type Database struct {
-	// Instructs the Operator to deploy a dedicated database.
+	// Instructs the operator to deploy a dedicated database.
 	// By default, a dedicated PostgreSQL database is deployed as part of the Che installation.
 	// When `externalDb` is set as `true`, no dedicated database is deployed by the
-	// Operator and you need to provide connection details to the external DB you want to use.
+	// operator and you need to provide connection details to the external database you want to use.
```
Comment (rolfedh) [inline api/v2/checluster_types.go:234]:
```suggestion
	// Operator. You must provide the connection details of the external database you want to use.
```
- Consider stating this user action as a separate sentence. 
- It's unclear whether the action is to provide the information _to_ or _about_ the database. I think you mean about.

File: api/v2/checluster_types.go [hunk 4]
```diff
@@ -399,7 +399,7 @@ type Proxy struct {
 
 // PersistentVolumeClaim custom settings.
 type PVC struct {
-	// Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.
+	// Persistent Volume Claim size. To update the claim size, the storage class that provisions it must support resizing.
```
Comment (rolfedh) [inline api/v2/checluster_types.go:402]: It seems like many of these issues you're fixing show up throughout the repo.
Would it make sense to batch update them?
For example:
```
Results for "claim size, Storage class" in "che-operator"

/home/rdlugyhe/che-operator/api/v2/checluster_types.go:
  402:  Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.


/home/rdlugyhe/che-operator/config/crd/bases/org.eclipse.che_checlusters.yaml:
  1621:                                claim size, Storage class that provisions it must support
  2049:                                claim size, Storage class that provisions it must support


/home/rdlugyhe/che-operator/deploy/deployment/kubernetes/combined.yaml:
  1089:  Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.
  1447:  Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.


/home/rdlugyhe/che-operator/deploy/deployment/openshift/combined.yaml:
  1089:  Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.
  1447:  Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.


/home/rdlugyhe/che-operator/helmcharts/next/crds/checlusters.org.eclipse.che.CustomResourceDefinition.yaml:
  1084:  Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.
  1442:  Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.


/home/rdlugyhe/che-operator/bundle/next/eclipse-che-preview-openshift/manifests/org.eclipse.che_checlusters.yaml:
  1666:                                  the claim size, Storage class that provisions it must
  2101:                                  the claim size, Storage class that provisions it must


/home/rdlugyhe/che-operator/deploy/deployment/kubernetes/objects/checlusters.org.eclipse.che.CustomResourceDefinition.yaml:
  1084:  Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.
  1442:  Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.


/home/rdlugyhe/che-operator/deploy/deployment/openshift/objects/checlusters.org.eclipse.che.CustomResourceDefinition.yaml:
  1084:  Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.
  1442:  Persistent Volume Claim size. To update the claim size, Storage class that provisions it must support resize.

```

File: api/v2/checluster_types.go (hunk 1 above)
Comment (deerskindoll) [reply api/v2/checluster_types.go:120]: I see. I thought it's capitalized only when it's a part of a full name ( XYZ Operator). Let me fix it.

File: api/v2/checluster_types.go (hunk 4 above)
Comment (deerskindoll) [reply api/v2/checluster_types.go:402]: I only work in the .go file in this particular repo. No idea what's happening in the other files. Let me ask.

Comment (tolusha) [reply api/v2/checluster_types.go:402]: @deerskindoll 
The .go file is a source of truth. Other files are updated automatically after running `make update-dev-resources` command.
So, submit changes for `checluster_types.go` and I will do the rest.

**Pull request merged**